internal/services/redhatopenshift/redhatopenshift_cluster_resource.go

```diff
@@ -0,0 +1,889 @@
+	if profile.SubnetID != nil {
+		subnetId = *profile.SubnetID
+	}
+
+	encryptionAtHostEnabled := profile.EncryptionAtHost == redhatopenshift.EncryptionAtHostEnabled
+
+	diskEncryptionSetId := ""
+
+	if profile.DiskEncryptionSetID != nil {
+		diskEncryptionSetId = *profile.DiskEncryptionSetID
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"vm_size":                    profile.VMSize,
+			"subnet_id":                  subnetId,
+			"encryption_at_host_enabled": encryptionAtHostEnabled,
+			"disk_encryption_set_id":     diskEncryptionSetId,
+		},
+	}
+}
+
+func flattenOpenShiftWorkerProfiles(profiles *[]redhatopenshift.WorkerProfile) []interface{} {
+	if profiles == nil {
+		return []interface{}{}
+	}
+
+	results := make([]interface{}, 0)
+
+	result := make(map[string]interface{})
+	result["node_count"] = int32(len(*profiles))
+
+	for _, profile := range *profiles {
+		if result["disk_size_gb"] == nil && profile.DiskSizeGB != nil {
+			result["disk_size_gb"] = profile.DiskSizeGB
+		}
+
+		if result["vm_size"] == nil && profile.VMSize != nil {
+			result["vm_size"] = profile.VMSize
+		}
+
+		if result["subnet_id"] == nil && profile.SubnetID != nil {
+			result["subnet_id"] = profile.SubnetID
+		}
+
+		if result["encryption_at_host_enabled"] == nil {
+			result["encryption_at_host_enabled"] = profile.EncryptionAtHost == redhatopenshift.EncryptionAtHostEnabled
+		}
+
+		if result["disk_encryption_set_id"] == nil && profile.DiskEncryptionSetID != nil {
+			result["disk_encryption_set_id"] = profile.DiskEncryptionSetID
+		}
+	}
+
+	results = append(results, result)
+
+	return results
+}
+
+func flattenOpenShiftAPIServerProfile(profile *redhatopenshift.APIServerProfile) []interface{} {
+	if profile == nil {
+		return []interface{}{}
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"visibility": string(profile.Visibility),
+		},
+	}
+}
+
+func flattenOpenShiftIngressProfiles(profiles *[]redhatopenshift.IngressProfile) []interface{} {
+	if profiles == nil {
+		return []interface{}{}
+	}
+
+	results := make([]interface{}, 0)
+
+	for _, profile := range *profiles {
+		result := make(map[string]interface{})
+		result["visibility"] = string(profile.Visibility)
+
+		results = append(results, result)
+	}
+
+	return results
+}
+
+func expandOpenshiftClusterProfile(
+	input []interface{},
+	subscriptionId string,
+) *redhatopenshift.ClusterProfile {
+	resourceGroupName := fmt.Sprintf("aro-%s", randomDomainName)
+	resourceGroupId := ResourceGroupID(subscriptionId, resourceGroupName)
+
+	if len(input) == 0 {
+		return &redhatopenshift.ClusterProfile{
+			ResourceGroupID:      utils.String(resourceGroupId),
+			Domain:               utils.String(randomDomainName),
+			FipsValidatedModules: redhatopenshift.FipsValidatedModulesDisabled,
+		}
+	}
+
+	config := input[0].(map[string]interface{})
+
+	pullSecret := config["pull_secret"].(string)
+
+	domain := config["domain"].(string)
+	if domain == "" {
+		domain = randomDomainName
+	}
```

Comment:
this should be set in the create where it is randomly created if the user doesn't specify it not here as this is shared with update. and is there an argument to just make it required? or is the random string how the portal/service behaves?

Reply:
it is mandatory field in azure portal.

Reply:
ah well then if its mandatory in the portal we should just make it mandatory in terraform too and ensure the user picks the name rather then randomly generating it